runtime/src/main/java/org/corfudb/runtime/collections/PersistentCorfuTable.java

```diff
@@ -0,0 +1,121 @@
+package org.corfudb.runtime.collections;
+
+import io.vavr.collection.HashMap;
+import org.corfudb.runtime.object.ICorfuSMR;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public class PersistentCorfuTable<K, V> implements ICorfuTable<K, V>, ICorfuSMR<PersistentCorfuTable<K, V>> {
+
+    private PersistentHashMapWrapper<K, V> mainMap;
+
+    public PersistentCorfuTable() {
```

Comment:
As I know Vjeko has already implemented a PersistentCorfu or something like that